setup.py

```diff
@@ -27,26 +28,40 @@
 with open("requirements.txt", "rt", encoding='utf8') as fh:
     requirements = fh.read().split('\n')
 
-setuptools.setup(
-    name="charmcraft",
-    version=charmcraft.__version__,
-    author="Facundo Batista",
-    author_email="facundo.batista@canonical.com",
-    description="The main tool to build, upload, and develop in general the Juju charms.",
-    long_description=long_description,
-    long_description_content_type="text/markdown",
-    url="https://github.com/canonical/charmcraft",
-    license="Apache-2.0",
-    packages=['charmcraft', 'charmcraft.commands', 'charmcraft.commands.store'],
-    classifiers=[
-        "Environment :: Console",
-        "License :: OSI Approved :: Apache Software License",
-        "Operating System :: OS Independent",
-        "Programming Language :: Python :: 3",
-    ],
-    entry_points={
-        'console_scripts': ["charmcraft = charmcraft.main:main"],
-    },
-    python_requires='>=3',
-    install_requires=requirements,
-)
+version_path = Path("charmcraft/version.py")
+version_backup = Path("charmcraft/version.py~")
+version_path.rename(version_backup)
+try:
+    with version_path.open("wt", encoding="utf8") as fh:
+        fh.write('''\
```

Comment:
Please move this text outside (to module level) to a "template" or something, so it doesn't break visual indentation. Thanks!

Reply:
I did it differently, please confirm it's alright :-)